tests/ui/associated-consts/associated-const-in-trait.stderr

```diff
@@ -30,6 +30,22 @@ LL |     const N: usize;
    |           ^ ...because it contains this associated `const`
    = help: consider moving `N` to another trait
 
-error: aborting due to 2 previous errors
+error[E0038]: the trait `Trait` is not dyn compatible
```

Comment:
we could store `not_dyn_compatible_traits: HashSet<DefId>` in the infcx to dedup these errors per body.